titus-server-master/src/main/java/com/netflix/titus/master/scheduler/opportunistic/OpportunisticCpuAvailability.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2019 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.titus.master.scheduler.opportunistic;
+
+import java.time.Instant;
+
+import com.netflix.titus.common.util.time.Clock;
+
+public class OpportunisticCpuAvailability {
+    private final String allocationId;
+    private final Instant expiresAt;
```

Comment:
`long` for better performance?

Reply:
I really prefer the readability and integration with the rest of the codebase, since it is not on the super hot path of the scheduling loop (code that uses it is only triggered when both the task and the agent match opportunistic requirements), would you be OK if we try it like this first, and optimize later as necessary?